src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java

```diff
@@ -105,12 +105,10 @@ public Response updateFinalAccessVote(Vote rec,
             Vote vote = api.firstVoteUpdate(rec, id);
             Document access = accessRequestAPI.describeDataAccessRequestById(requestId);
             List<Integer> dataSets = DarUtil.getIntegerList(access, DarConstants.DATASET_ID);
-            if(access.containsKey(DarConstants.RESTRICTION)){
-                List<Vote> votes = vote.getType().equals(VoteType.FINAL.getValue()) ? api.describeVoteByTypeAndElectionId(VoteType.AGREEMENT.getValue(), vote.getElectionId()) :  api.describeVoteByTypeAndElectionId(VoteType.FINAL.getValue(), vote.getElectionId());
-                if(vote.getVote() != null && votes.get(0).getVote() != null){
-                    electionAPI.updateFinalAccessVoteDataRequestElection(rec.getElectionId());
-                }
-            }else {
+            List<Vote> votes = vote.getType().equals(VoteType.FINAL.getValue()) ?
+                api.describeVoteByTypeAndElectionId(VoteType.AGREEMENT.getValue(), vote.getElectionId()) :
```

Comment:
Maybe I am interpreting this wrong, but if the vote type is Final then why pass in the Agreement vote type on the next line, instead of passing in Final?

Reply:
My intention here was to leave the original logic in place and do as little refactoring as possible that might cause a regression. However, after reviewing what this code is doing, the end result is the same regardless of what votes we query for, either final or agreement. For example, in both cases, we end up calling `electionAPI.updateFinalAccessVoteDataRequestElection(rec.getElectionId());` This code can be simplified even further, nice call 👍🏽

Reply:
I've rewritten the logic here to accomplish the same thing, but hopefully in a more code-legible manner. Please have a second look.